File: sorting my life (sort algarithoms)/stalinsort.py
datatosort = [3, 6, 8, 10, 1, 2, 1,42,15,57]

def Stalin_sort(array):
    max = array[0]
    new_array = [max]
    for i in array[1:]:
        if i >= max:
            max = i
            new_array.append(i)
    return new_array


    


print(Stalin_sort(datatosort))

File: sorting my life (sort algarithoms)/stalinsort.js
let datatosort = [3, 6, 8, 10, 1, 2, 1,42,15,57]

function stalinSort(arr) {
    let sorted = [arr[0]]
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] >= sorted[sorted.length - 1]) {
            sorted.push(arr[i])
        }
    }
    return sorted
}

console.log(stalinSort(datatosort)) // [3, 6, 8, 10, 42, 57]
